src/stream/src/task/stream_manager.rs

```diff
@@ -185,17 +194,25 @@ impl LocalStreamManager {
     /// Use `epoch` to find collect rx. And wait for all actor to be collected before
     /// returning.
     pub async fn collect_barrier(&self, epoch: u64) -> CollectResult {
-        let rx = {
+        let (rx, timer) = {
             let core = self.core.lock();
             let mut barrier_manager = core.context.lock_barrier_manager();
             barrier_manager.remove_collect_rx(epoch)
         };
         // Wait for all actors finishing this barrier.
-        rx.await.unwrap()
+        let result = rx.expect("no rx for local mode").await.unwrap();
+        timer.expect("no timer for test").observe_duration();
```

Comment:
In the current implementation, we start timer for `barrier_inflight_latency` when CN receives inject barrier request and call send_barrier and stop the timer when all actors finishing collecting the barrier. I am confused what `barrier_inflight_latency` measures: it is neither the e2e latency for a barrier to flow through the graph, nor the latency for a barrier to flow through local CN. Looks like it is measuring the latency of the barrier flowing through a partial graph.